### PR TITLE
Sort saved filters by alphabetical order

### LINKED ISSUE #1284: [Bug]: saved filters are not sorted
### Verified issue does not already exist?

- [X] I have searched and found no existing issue

### What happened?

I love the new saved filters! So much so that I have already created thirteen of them. However, these remain listed in the order that I created them. It would be ideal if they could list in alphabetical order.

### What error did you receive?

Saved filters are not listed in a logical order

### Where are you hosting Actual?

Fly.io

### What browsers are you seeing the problem on?

Safari

### Operating System

Mac OSX

### PR DESCRIPTION
**Pull request by aleetsaiya**
fix #1284

### CODE REVIEW

**Comment by Kidglove57** [comment]
Thank you for responding to my plea! I have tested it with my own budget and my thirteen saved filters. It certainly solves the issue from my lay perspective.

File: packages/desktop-client/src/components/autocomplete/SavedFilterAutocomplete.js
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/SavedFilterAutocomplete.js:50]: **💬 suggestion:** ‏would you mind doing the sorting inside of `useFilters`? Then the fix would be available in all the places where this hook is used.

Also: can you make sure to use `useMemo` here? We should sort only once, not on every render cycle

**Review by MatissJanis — COMMENTED** [review]
👋 Thanks for the patch! Love the initiative and the PRs you've been sending.

But would you mind just changing it a bit to generalize the fix?

**Comment by aleetsaiya** [comment]
I did some research to know which parts of the UI would be affected, and I found the filter menu is also sorted.

Before:
截圖 2023-07-08 上午3 27 51

After:
截圖 2023-07-08 上午4 18 08

**Comment by MatissJanis** [comment]
> I did some research to know which parts of the UI would be affected when I add the sort logic inside `useFilters`, and I found the filter menu is also sorted.
> 
> Before: 截圖 2023-07-08 上午3 27 51
> 
> After: 截圖 2023-07-08 上午4 18 08

That's cool actually. Makes for a consistent UX.

**Review by MatissJanis — APPROVED** [review]
Looks good on my end!

(not merging just yet as we're in a soft code-freeze; we want to solve some more bugs and release v23.7.2; after that we plan to unblock the merges again)

**Pull request merged**
